db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -864,7 +864,7 @@ public CostEstimate estimateCost(OptimizablePredicateList predList,
                     if(baseTableRestrictionList.isRedundantPredicate(i)) continue;
                 }
 
-                if(!p.isJoinPredicate() || currentJoinStrategy.allowsJoinPredicatePushdown()) //skip join predicates unless they support predicate pushdown
+                if(!p.isJoinPredicate()&& !p.isFullJoinPredicate() || currentJoinStrategy.allowsJoinPredicatePushdown()) //skip join predicates unless they support predicate pushdown
```

Comment:
Thanks @msirek . Here is my understanding of the current scope of isJoinPredicate(): it is a binary relational operator where each side references a different table. So it does apply to the full join case too.  

Under the existing implementation, all conditions in the ON clause are pushed to the right source of a join and put in the same list with the filtering single table conditions of the right source if any.

The logic above before this PR is to select among the combined list the filtering single table conditions to compute selectivity and do cost estimation. 
With full outer join, single table condition is no longer necessarily filtering conditions, it could be from the ON clause, for example:
```
select 1 from t1 full join t2 on b2=1;
```
This is the reason we add the extra check to rule that out.

Reply:
Oh, I see, isJoinPredicate() is just checking that the predicate references at least two different tables.  We could have a full outer join where the predicate only references a single table.  The method naming is just a little confusing.  isJoinPredicate should maybe be predReferencesTwoOrMoreTables, since it is possible to have a predicate be a join predicate when isJoinPredicate() returns false.